### DISCUSSION TITLE
2023-06-29 - kampus coworking daily notes

### DISCUSSION THREAD

**Discussion by usirin** [kamp-us/pano#469] [General]
Things to do, notes from stream, etc etc

**Comment by usirin** [comment]
add planetscale

**Reply by usirin** [reply]
i knew branching idea was cool, but i, *again*, came to the conclusion without having the enough information.

it allows us to branch out from our "Prod" db in every pr preview :corefuWOA2:

**Reply by usirin** [reply]
prisma & planetscale best practices requires us to not use prisma migrations when deploying to pscale.
that's fine, we can still use migrations in dev, then our `ci/cd` pipeline would run the `prisma db push`.

**Comment by usirin** [comment]
monorepo stuff

**Reply by usirin** [reply]
went through the pull requests, merged #466. LETS GOOOOOOO @akselsaatci congratz on your first PR

**Reply by usirin** [reply]
Another "First Pull Request Merged" achievement from @alicangunduz with #470. LETS GOOOOOOOOOOO

**Reply by akselsaatci** [reply]
LETSS GOOOOOOOOO 🎉🎉🎉

**Comment by usirin** [comment]
search????

**Reply by usirin** [reply]
what are our options?

I only know algolia 😅

**Reply by usirin** [reply]
@Ketcap @EmirBoyaci @puskuruk any ideas?

**Reply by puskuruk** [reply]
I need to know exactly what we need to do to answer this. What do we mean by `search`? What is the size of the data that we're going to search for? Do we need full-text search? Do we want to search in a database, or do we have a JSON file? etc. I mostly don't believe that we don't need a third-party solution. If we're going to implement that search functionality to search in our database, my solution would be to use PostgreSQL's native solutions.

**Reply by puskuruk** [reply]
Since we use Prisma and MySQL(PlanetScale) we can check the discussion below. Also [Prisma explains how we can use it in their official docs](https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search)

[https://github.com/prisma/prisma/discussions/11249](https://github.com/prisma/prisma/discussions/11249)

**Reply by Ketcap** [reply]
+1 with @puskuruk , prisma has good enough full text search for now. Then when it get bigger we can go similiar to Algolia or even Algolia :D

**Reply by EmirBoyaci** [reply]
I don't know exactly what we're mentioning by "search" but I know [meilisearch](https://www.meilisearch.com/) which is a flexible and powerful user-focused search engine.

> Since we use Prisma and MySQL(PlanetScale) we can check the discussion below. Also [Prisma explains how we can use it in their official docs](https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search)
> 
> [prisma/prisma#11249](https://github.com/prisma/prisma/discussions/11249)

^ I agree with @puskuruk we can start by using Prisma's text search support. In the future, if we came up some problems like latency and so on, we may move to the another possible solution tho.

**Reply by usirin** [reply]
We have pano with db, yes. But sozluk's content is being generated from mdx files and being fed to the gql via `contentlayer`, which means we don't have a db there. This requires us to have a higher level search abstraction which defers to the lower level search functionalities. (fulltextsearch via db, and in-memory search solution for sozluk for now)

I mostly agree about not using a third party but, the above situation is making me think that maybe we should start doing it from the beginning.

**Reply by Ketcap** [reply]
@usirin what about turning content into vectors and do json vector search ? convertion on vector can be done on build time
